kyaml/runfn/runfn.go

```diff
@@ -468,11 +468,24 @@ func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser
 		if err != nil {
 			return nil, err
 		}
+
+		// Storage mounts can either come from kustomize fn run --mounts,
+		// or from the declarative function mounts field.
+		// In the latter case, we ensure that the mount path is subject
+		// to kustomize's load restrictions.
+		storageMounts := spec.Container.StorageMounts
+		for _, mount := range storageMounts {
+			if strings.HasPrefix(filepath.Clean(mount.Src), "../") {
+				return nil, errors.Errorf("mount path %s is not permitted; declarative mount paths must be under the current directory", mount.Src)
```

Comment:
I was just sanity checking against the Starlark file version of this constraint, and it handles another case: absolute paths. If we don't ban those, we'll need to be comparing more specifically to the kustomize root.

Reply:
Dug a little bit and I think we have to allow absolute paths. With a relative path, I get this error from docker:

`docker: Error response from daemon: invalid mount config for type "bind": invalid mount path: 'testdata/charts' mount path must be absolute.`

I'll update this check here to test for absolute paths.

Reply:
Updated!